### PR TITLE
docs: add pointfree `alt` and revise pointfree docs

### LINKED ISSUE #1286: doct: Alt documentation
# Bug report

## What's wrong

There's not a section for the alt function under Composition helpers -> Pointfree
Screen Shot 2022-02-17 at 9 17 14

## How is that should be

It would be helpful to have a section describing it and showing its usage with an example like the following:
Screen Shot 2022-02-17 at 9 17 53

### PR DESCRIPTION
**Pull request by ryangrose**
# I have made things!

- Added docs for `alt` in pointfree.rst
- Revised pointfree docs for clarity.
  - If this is helpful I can apply similar changes to docstrings to make those more clear and keep them consistent

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #1286

### CODE REVIEW

File: docs/pages/pointfree.rst [hunk 1]
```diff
@@ -33,69 +34,118 @@ Because they do not compose well:
...
 
-  >>> def bindable(arg: str) -> Maybe[int]:
-  ...     return Some(1)
+  >>> def index_of_1(arg: str) -> Maybe[int]:
+  ...     if '1' in arg:
+  ...         return Some(arg.index('1'))
+  ...     return Nothing
 
-  >>> container: Maybe[str] = Some('a')
+  >>> container: Maybe[str] = Some('A1 Steak Sauce')
```
Comment (sobolevn) [inline docs/pages/pointfree.rst:142]:
```suggestion
  >>> container = Some('A1 Steak Sauce')
```

**Review by sobolevn — COMMENTED** [review]
Wow, awesome work! 👍 
It really helps our docs.

**Review by sobolevn — APPROVED** [review]
Thank you!

**Comment by sobolevn** [comment]
If you have the time - you can fix RTD builds in the next PR :)

**Comment by ryangrose** [comment]
> If you have the time - you can fix RTD builds in the next PR :)

Yeah I was trying to look into it but couldn't figure it out? [RTD says it's missing tomlkit](https://readthedocs.org/projects/returns/builds/18038084/) but it looks like it's in the pyproject.toml. Any ideas?

The only thing I can think of is that poetry install isn't installing to the same python that RTD is using. Maybe [export requirements.txt](https://python-poetry.org/docs/cli/#export) as a pre-install step then list that as the RTD install requirements?

**Comment by ryangrose** [comment]
Created #1481 to track it

> > If you have the time - you can fix RTD builds in the next PR :)
> 
> Yeah I was trying to look into it but couldn't figure it out? [RTD says it's missing tomlkit](https://readthedocs.org/projects/returns/builds/18038084/) but it looks like it's in the pyproject.toml. Any ideas?
> 
> The only thing I can think of is that poetry install isn't installing to the same python that RTD is using. Maybe [export requirements.txt](https://python-poetry.org/docs/cli/#export) as a pre-install step then list that as the RTD install requirements?

**Pull request merged**